app/controllers/api/consignments_controller.rb

```diff
@@ -5,22 +5,42 @@ class ConsignmentsController < RestController
     before_action :require_authentication
 
     def update_price
-      artworks_price = fetch_sale_artworks_with_price(params[:sale_id])
+      update_sale_info(params[:sale_id])
 
-      PartnerSubmissionService.update_price(artworks_price)
       render json: { result: 'ok' }, status: :created
     end
 
     private
 
-    def fetch_sale_artworks_with_price(sale_id)
+    def update_sale_info(sale_id)
+      sale = Gravity.client.sale(id: sale_id)._get
       sale_artworks =
         Gravity.client.sale_artworks(sale_id: sale_id).sale_artworks
 
       sale_artworks.map do |sale_artwork|
-        artwork_id = sale_artwork.artwork.id
+        artwork = sale_artwork.artwork
+        submission = Submission.find_by(source_artwork_id: artwork.id)
```

Comment:
**suggestion (non-blocking):** We should consider indexing the `source_artwork_id` column depending on the performance of this endpoint. I've added a [ticket](https://artsyproduct.atlassian.net/browse/SWA-194) to the backlog for this work.

Reply:
It sounds good